### DISCUSSION TITLE
understanding reason IPV4LL in hook scripts

### DISCUSSION THREAD

**Discussion by rat-dev** [rsmarples/dhcpcd#426] [Q&A]
Hi all! I try to better understand the IPv4LL transitions of dhcpcd, for 'perfect' hook handling (-:

What I see is that when no DHCP answer is received, with ipv4ll enabled the hooks are triggered with the following vars:

interface=media2
pid=1961
protocol=ipv4ll
reason=IPV4LL
ifcarrier=up
ifflags=4163
ifmetric=206
ifmtu=1500
ifwireless=0
old_gateway=
new_broadcast_address=169.254.255.255
new_gateway=
new_ip_address=169.254.91.227
new_network_number=169.254.0.0
Dnew_subnet_cidr=16
new_subnet_mask=255.255.0.0

After finally a lease was aquired and the interface bound to the IP, the following seems to indicated the Ipv4LL address  removal:

interface=media2
pid=1961
protocol=ipv4ll
reason=IPV4LL
ifcarrier=up
ifflags=4163
ifmetric=206
ifmtu=1500
ifwireless=0
old_gateway=
new_gateway=

My question is: how to reliably detect the IPv4LL removal after DHCP binding? is it enough that for with reason IPV4LL no new ip address is set? or  that both old and new address are missing? (and why are there gateway vars? )

Thank you very much,

Harry

**Comment by rsmarples** [comment]
That looks like a bug
old_ip_address should be 169.254.92.227 and new_ip_address should be blank
